dolphin/db/sqlalchemy/api.py

```diff
@@ -232,34 +236,49 @@ def storage_create(context, values):
     with session.begin():
         session.add(storage_ref)
 
-    return _storage_get(context,
-                        storage_ref['id'],
-                        session=session)
+    return _resource_get(context,
+                         storage_ref['id'],
+                         models.Storage,
+                         session=session)
 
 
 def storage_update(context, storage_id, values):
     """Update a storage device with the values dictionary."""
-    return NotImplemented
+    model = models.Storage
+    session = get_session()
+    with session.begin():
+        query = storage_get(context, storage_id)
+        query.update(values)
+        query.save(session=session)
+    return
 
 
 def storage_get(context, storage_id):
     """Retrieve a storage device."""
-    return _storage_get(context, storage_id)
+    return _resource_get(context, storage_id, models.Storage)
 
 
-def _storage_get(context, storage_id, session=None):
-    result = (_storage_get_query(context, session=session)
-              .filter_by(id=storage_id)
-              .first())
+def _resource_get_all(context, model, session=None):
+    result = (_resource_get_query(context, model, session=session)
+              .all())
+    if not result:
+        raise exception.StorageNotFound()
 
+    return result
+
+
+def _resource_get(context, resource_id, model, session=None):
```

Comment:
i think, we need to make a common function, as _resource_get() otherwise, for everything, new func need to write. for example: _pool_get(), _storage_get() or _volume_get() etc. Here once function will do the job